[hipblaslt] update CHANGELOG (#1880)
ROCm/rocm-libraries: projects/hipblaslt/CHANGELOG.md

Full documentation for hipBLASLt is available at [rocm.docs.amd.com/projects/hipBLASLt](https://rocm.docs.amd.com/projects/hipBLASLt/en/latest/index.html).

## hipBLASLt 1.1.0 for ROCm 7.1.0

### Added

* Fused Clamp GEMM in hipBLASLt (enabled by ``HIPBLASLT_EPILOGUE_CLAMP_EXT`` and ``HIPBLASLT_EPILOGUE_CLAMP_BIAS_EXT``, requiring the min (``HIPBLASLT_MATMUL_DESC_EPILOGUE_ACT_ARG0_EXT``) and max (``HIPBLASLT_MATMUL_DESC_EPILOGUE_ACT_ARG1_EXT``) to be set)
* Support ReLU/Clamp activation functions with auxiliary output for f16/bf16 data types for gfx942 to capture intermediate results (enabled by ``HIPBLASLT_EPILOGUE_RELU_AUX``, ``HIPBLASLT_EPILOGUE_RELU_AUX_BIAS``, ``HIPBLASLT_EPILOGUE_CLAMP_AUX_EXT``, and ``HIPBLASLT_EPILOGUE_CLAMP_AUX_BIAS_EXT``)
* Support SS_BSS datatype (``HIPBLAS_COMPUTE_32F_FAST_16BF``) for gfx950 only
* Added the CPP extension APIs ``setMaxWorkspaceBytes`` and ``getMaxWorkspaceBytes``
* Added the log print (`HIPBLASLT_LOG_MASK=32`) for Grouped Gemm
* Supported swizzleA by hipblaslt-ext cpp API

> * Support hipblaslt extop for gfx11xx/gfx12xx

Reviewer: ```suggestion
* Fused Clamp GEMM for ``HIPBLASLT_EPILOGUE_CLAMP_EXT`` and ``HIPBLASLT_EPILOGUE_CLAMP_BIAS_EXT``. This feature requires the minimum (``HIPBLASLT_MATMUL_DESC_EPILOGUE_ACT_ARG0_EXT``) and maximum (``HIPBLASLT_MATMUL_DESC_EPILOGUE_ACT_ARG1_EXT``) to be set.
* Support for ReLU/Clamp activation functions with auxiliary output for the `f16` and `bf16` data types for gfx942 to capture intermediate results. This feature is enabled for ``HIPBLASLT_EPILOGUE_RELU_AUX``, ``HIPBLASLT_EPILOGUE_RELU_AUX_BIAS``, ``HIPBLASLT_EPILOGUE_CLAMP_AUX_EXT``, and ``HIPBLASLT_EPILOGUE_CLAMP_AUX_BIAS_EXT``.
* Support for the ``SS_BSS`` data type (``HIPBLAS_COMPUTE_32F_FAST_16BF``) for gfx950 only.
* Added the CPP extension APIs ``setMaxWorkspaceBytes`` and ``getMaxWorkspaceBytes``.
* Added the ability to print logs (using ``HIPBLASLT_LOG_MASK=32``) for Grouped GEMM.
* Support for swizzleA by using the hipblaslt-ext cpp API.
* Support for hipBLASLt extop for gfx11xx and gfx12xx.
```

### Changed

> * If the workspace size is insufficient, ``hipblasLtMatmul()`` would return an error rather than causing a segmentation fault.

Reviewer: ```suggestion
* ``hipblasLtMatmul()`` now returns an error when the workspace size is insufficient, rather than causing a segmentation fault.
```

Full documentation for hipBLASLt is available at [rocm.docs.amd.com/projects/hipBLASLt](https://rocm.docs.amd.com/projects/hipBLASLt/en/latest/index.html).

## hipBLASLt 1.1.0 for ROCm 7.1.0

### Added

* Fused Clamp GEMM in hipBLASLt (enabled by ``HIPBLASLT_EPILOGUE_CLAMP_EXT`` and ``HIPBLASLT_EPILOGUE_CLAMP_BIAS_EXT``, requiring the min (``HIPBLASLT_MATMUL_DESC_EPILOGUE_ACT_ARG0_EXT``) and max (``HIPBLASLT_MATMUL_DESC_EPILOGUE_ACT_ARG1_EXT``) to be set)

> * Support ReLU/Clamp activation functions with auxiliary output for f16/bf16 data types to capture intermediate results (enabled by ``HIPBLASLT_EPILOGUE_RELU_AUX``, ``HIPBLASLT_EPILOGUE_RELU_AUX_BIAS``, ``HIPBLASLT_EPILOGUE_CLAMP_AUX_EXT``, and ``HIPBLASLT_EPILOGUE_CLAMP_AUX_BIAS_EXT``)

Reviewer 2: Please check if this is only for gfx942?

> * Support for the ``SS_BSS`` data type (``HIPBLAS_COMPUTE_32F_FAST_16BF``) for gfx950 only.

Reviewer 2: SS_BSS is not a known type for hipblaslt user.

Reviewer 2: I would recommend to rephrase it as :
Support for ``HIPBLAS_COMPUTE_32F_FAST_16BF`` for FP32 data type for gfx950 only.
@amd-jnovotny  any comment?

Reviewer: Yes, that sounds good. I'd put FP32 in code ticks because it's a data type.

* Added the CPP extension APIs ``setMaxWorkspaceBytes`` and ``getMaxWorkspaceBytes``.
* Added the ability to print logs (using ``HIPBLASLT_LOG_MASK=32``) for Grouped GEMM.
* Support for swizzleA by using the hipblaslt-ext cpp API.
* Support for hipBLASLt extop for gfx11xx and gfx12xx.

> * Support ldd and ldc in MBSK

Reviewer 2: we need to fix this one

Full documentation for hipBLASLt is available at [rocm.docs.amd.com/projects/hipBLASLt](https://rocm.docs.amd.com/projects/hipBLASLt/en/latest/index.html).

## hipBLASLt 1.1.0 for ROCm 7.1.0

### Added

* Fused Clamp GEMM in hipBLASLt (enabled by ``HIPBLASLT_EPILOGUE_CLAMP_EXT`` and ``HIPBLASLT_EPILOGUE_CLAMP_BIAS_EXT``, requiring the min (``HIPBLASLT_MATMUL_DESC_EPILOGUE_ACT_ARG0_EXT``) and max (``HIPBLASLT_MATMUL_DESC_EPILOGUE_ACT_ARG1_EXT``) to be set)
* Support ReLU/Clamp activation functions with auxiliary output for f16/bf16 data types to capture intermediate results (enabled by ``HIPBLASLT_EPILOGUE_RELU_AUX``, ``HIPBLASLT_EPILOGUE_RELU_AUX_BIAS``, ``HIPBLASLT_EPILOGUE_CLAMP_AUX_EXT``, and ``HIPBLASLT_EPILOGUE_CLAMP_AUX_BIAS_EXT``)
* Support SS_BSS datatype (``HIPBLAS_COMPUTE_32F_FAST_16BF``) for gfx950 only
* Added the CPP extension APIs ``setMaxWorkspaceBytes`` and ``getMaxWorkspaceBytes``

> * Added the log print for Grouped Gemm

Reviewer 2: add (HIPBLASLT_LOG_MASK=32) into this item

* Supported swizzleA by hipblaslt-ext cpp API

> * Completed the support of `--api_method` mix and cpp with `--swizzleA` for hipblaslt-bench

Reviewer 2: this item is duplicated with "Supported swizzleA by hipblaslt-ext cpp API". Please remove it

> * Support ldd and ldc in MSKB

Reviewer 2: please check it with Josh about what the impact is to user
